opener/src/windows/reveal.rs

```diff
@@ -22,57 +17,13 @@ pub(crate) fn reveal(path: &Path) -> Result<(), OpenError> {
 }
 
 fn reveal_in_thread(path: &Path) -> io::Result<()> {
-    to_io_result(unsafe { CoInitializeEx(ptr::null_mut(), COINIT_MULTITHREADED) })?;
-    let item_id_list = ItemIdList::new(path)?;
-    // Because the cidl argument is zero, SHOpenFolderAndSelectItems opens the singular item
-    // in our item id list in its containing folder and selects it.
-    to_io_result(unsafe { SHOpenFolderAndSelectItems(item_id_list.0, 0, ptr::null(), 0) })?;
+    unsafe { CoInitializeEx(None, COINIT_MULTITHREADED) }?;
+    // The ILCreateFromPathW function expects a canonicalized path.
+    // Unfortunately it does not support NT UNC paths (which std::path::canonicalize returns)
+    // so we use the normpath crate instead.
+    let path = convert_path(path.normalize()?.as_os_str())?;
+    let item_id_list = unsafe { ILCreateFromPathW(PCWSTR::from_raw(path.as_ptr())) };
+    unsafe { SHOpenFolderAndSelectItems(item_id_list, None, 0) }?;
```

Comment:
This is one of those things I'm not sure of: do we still need to call `ILFree` on `item_id_list` after `SHOpen...`?

Reply:
Yes, and it needs to be called even if `SHOpenFolderAndSelectItems()` fails. The same goes for `CoUninitialize()`.